docker/wrapper.py

```diff
@@ -12,37 +12,131 @@
+
+from shutil import copyfile
+#import ConfigParser
+
+import socket
 
 logging.basicConfig(level=logging.INFO)
 log = logging.getLogger()
 
-
 def call_pipeline(mount, args):
-    work_dir = os.path.join(mount, 'Toil-RNAseq-' + str(uuid4()))
+
+    #Activate the virtual environment where the toil rnaseq pipeline is installed
+    activate_this = '/opt/rnaseq-pipeline/toil_venv/bin/activate_this.py'
+    execfile(activate_this, dict(__file__=activate_this))
+
+    if args.auto_scale:
+        #Run the meoso-master process
+        #Run it in the background by using Popen instead of check_call 
+        #https://stackoverflow.com/questions/32577071/execute-subprocess-in-background
+        command = ('mesos-master --log_dir=/var/lib/mesos --registry=in_memory --cluster=' + args.cluster_name)
+        command = command.split()
+        try:
+            log.info('Executing mesos command: ' + str(command))
+            p = subprocess.Popen(command)
+        except OSError as e:
+            print(e.message, file=sys.stderr)
+            exit(e.returncode)
+        else:
+            log.info('mesos-master process running in background')
+
+
+    unique_toil_dir = 'Toil-RNAseq-' + str(uuid4())
+    work_dir = os.path.join(mount, unique_toil_dir)
+
     os.makedirs(work_dir)
     log.info('Temporary directory created: {}'.format(work_dir))
     config_path = os.path.join(work_dir, 'toil-rnaseq.config')
-    job_store = os.path.join(args.resume, 'jobStore') if args.resume else os.path.join(work_dir, 'jobStore')
+
+    output_dir = mount
+    if args.auto_scale:
+        #set job store to be clould URL, e.g. S3 bucket
+        job_store = os.path.join(args.resume, 'jobStore') if args.resume else args.job_store
+        output_dir = args.output_location
+    else:
+        job_store = os.path.join(args.resume, 'jobStore') if args.resume else os.path.join(work_dir, 'jobStore')
+   
+
     with open(config_path, 'w') as f:
-        f.write(generate_config(args.star, args.rsem, args.kallisto, mount,
+        f.write(generate_config(args.star, args.rsem, args.kallisto, args.hera, output_dir,
                                 args.disable_cutadapt, args.save_bam, args.save_wiggle,
-                                args.bamqc))
+                                args.bamqc, args.max_sample_size))
     loglevel = log.getEffectiveLevel()
 
-    command = ['toil-rnaseq', 'run',
+    if args.auto_scale:
+        #Pick the working directory carefully; it must exist on the workers
+        #Do not let Toil pick the directory because this will not work when
+        #launching the workflow with Dockstore because dockstore uses cwltool
+        #which makes the file system read only which will prevent Toil from 
+        #creating the directory. We cannot use a directory like
+        #/home/ubunut/... becuase that does not exist on coreos which is what
+        #the worker nodes use for an OS; '/tmp' will usually already be 
+        #created on most Linux distributions
```

Comment:
Again, thanks for including these comments. I'd check for the existence of `/tmp` so you can throw a specific error if it's not there for some odd reason.

Reply:
Not sure how to check if /tmp exists because this would be on a worker node that will be created by the Toil provisioner.

Reply:
Ah, good point — `/tmp` should be fine then.